### PR TITLE
Deployment of CephFS-Mirror.

### PR DESCRIPTION
**Pull request by hkadam134**
Adding cephfs-mirror deployment Code for Tier1.

Logs : http://magna002.ceph.redhat.com/ceph-qe-logs/hemanth/cephci-run-CDE35O/

### CODE REVIEW

File: suites/pacific/cephfs/tier_1_cephfs_mirror.yaml [hunk 1]
```diff
@@ -0,0 +1,141 @@
...
+                    - cephfs                        # name of the filesystem
+                  args:
+                    placement:
+                      nodes:
+                        - node4
+                        - node5
+      desc:  CephFS Mirror cluster deployment using cephadm
+      destroy-clster: false
+      module: test_cephadm.py
+      name: deploy cephfs-mirror
```
Comment (psathyan) [inline suites/pacific/cephfs/tier_1_cephfs_mirror.yaml:141]: Could we add a polarion ID? This is a new deployment... helps keep track with test cases

File: suites/pacific/cephfs/tier_1_cephfs_mirror.yaml [hunk 2]
```diff
@@ -0,0 +1,141 @@
...
+#    Each cluster configuration
+#    6-Node cluster(RHEL-8.3 and above)
+#    3 MONS, 2 MGR, 2 MDS, 3 OSD and 1 CEPHFS MIRROR, 1 Client service daemon(s)
+#     Node1 - Mon, Mgr, Installer
+#     Node2 - Mon, Mgr
+#     Node3 - Mon, OSD
+#     Node4 - OSD,MDS
+#     Node5 - OSD, MDS
+#     Node6 - CephFS Mirror
+#     Node7 - Client
```
Comment (psathyan) [inline suites/pacific/cephfs/tier_1_cephfs_mirror.yaml:17]: Cluster 2 has no node with id `7`

**Review by psathyan — COMMENTED** [review]
Looks good... it would be great if you could add the Polarion ID for the cluster deployment test case.

File: suites/pacific/cephfs/tier_1_cephfs_mirror.yaml (hunk 2 above)
Comment (hkadam134) [reply suites/pacific/cephfs/tier_1_cephfs_mirror.yaml:17]: We don't need cephfs-mirror node on cluster2 ,

May be I can mention as "#     Node6 - CephFS Mirror (Only on 1st cluster)"

Comment (sunilkumarn417) [reply suites/pacific/cephfs/tier_1_cephfs_mirror.yaml:17]: IMO, Splitting up the cluster conf would be good in the comments.

File: ceph/ceph_admin/cephfs_mirror.py [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+
+
+class CephfsMirror(ApplyMixin, Orch):
+    """Interface to ceph orch <action> rbd-mirror."""
+
+    SERVICE_NAME = "cephfs-mirror"
+
+    def apply(self, config: Dict) -> None:
+        """
+        Deploy the cephfs-mirror service using the provided configuration.
```
Comment (sunilkumarn417) [inline ceph/ceph_admin/cephfs_mirror.py:15]: please follow the doc string as used in this method.
https://github.com/red-hat-storage/cephci/blob/master/ceph/ceph_admin/daemon.py#L13-L31

File: suites/pacific/cephfs/tier_1_cephfs_mirror.yaml (hunk 2 above)
Comment (hkadam134) [reply suites/pacific/cephfs/tier_1_cephfs_mirror.yaml:16]: Done.

**Review by psathyan — APPROVED** [review]
Looks good to me

**Review by sunilkumarn417 — APPROVED** [review]

**Pull request merged**